### DISCUSSION TITLE
Injecting subgraph configuration - How to modify Query Field after it has been created?

### DISCUSSION THREAD

**Discussion by 2snEM6** [hayes/pothos#992] [Q&A] [answered]
In my GraphQL project, we have multiple subgraphs that use the same types and models.

Each subgraph is scoped in its own folder. Such as:

```
.
├── subgraphs/
│   ├── subgraph_1/
│   │   ├── query_field_1.ts
│   │   └── query_field_n.ts
│   └── subgraph_n/
│       ├── query_field_1.ts
│       └── query_field_n.ts
└── builder.ts
```

Each subgraph imports `builder.ts` and uses it to add query fields. Each query field (as seen above) is in its own file. This works well. 

## **The problem**

I don't want each query field to have to specify the subgraph it belongs to. It is tedious and prone to human errors.
## **What I thought of doing**

I thought of building a higher abstraction over the `builder.queryField` method and inject the `t.subGraphs` property.

However this is not possible due to the thunk pattern. I also thought of modifying the field after it has been created and manually setting the `subgraphs` property, but this seems not to be possible.

Any ideas 💡?

Thanks!

**Comment by hayes** [comment]
This will become easier in Pothos v4.  I don't think there is currently a good way to do this that isn't a hack

**Reply by 2snEM6** [reply, marked as answer]
I got it working by overriding queryField and doing something like this:

```typescript
queryField(
    name: string,
    field: QueryFieldThunk<PothosSchemaTypes.ExtendDefaultTypes<Types>>,
  ): void {
    const builtField = field(new QueryFieldBuilder(this));
    this.configStore.addFields('Query', () => ({
      [name]: builtField,
    }));

    if (this.currentSubgraph) {
      this.fieldRefSubgraphWeakMap.set(builtField, this.currentSubgraph);
    }

    this.configStore.onFieldUse(builtField, (fieldConfig) => {
      const subgraph = this.fieldRefSubgraphWeakMap.get(builtField);
      if (subgraph) {
        // eslint-disable-next-line no-param-reassign
        fieldConfig.pothosOptions.subGraphs = [subgraph];
      }
    });
  }
  ```

**Reply by 2snEM6** [reply]
Seems hacky yeah but works wonders
